fe/fe-core/src/main/java/com/starrocks/lake/LakeTable.java

```diff
@@ -110,7 +116,58 @@ public void write(DataOutput out) throws IOException {
     public static LakeTable read(DataInput in) throws IOException {
         // type is already read in Table
         String json = Text.readString(in);
-        LakeTable table = GsonUtils.GSON.fromJson(json, LakeTable.class);
-        return table;
+        return GsonUtils.GSON.fromJson(json, LakeTable.class);
+    }
+
+    @Override
+    public void onDrop(Database db, boolean force, boolean replay) {
+        dropAllTempPartitions();
+    }
+
+    @Override
+    public Runnable delete(long dbId, boolean replay) {
+        GlobalStateMgr.getCurrentState().getLocalMetastore().onEraseTable(dbId, this, replay);
+        return new DeleteTableTask(this, replay);
+    }
+
+    private static class DeleteTableTask implements Runnable {
+        private final LakeTable table;
+        private final boolean replay;
+
+        DeleteTableTask(LakeTable table, boolean replay) {
+            this.table = table;
+            this.replay = replay;
+        }
+
+        @Override
+        public void run() {
```

Comment:
This method was extracted from the original method `LocalMetaStore.onEraseOlapOrLakeTable()`